contracts/governance/extensions/GovernorTimelockAccess.sol

```diff
@@ -0,0 +1,216 @@
+
+pragma solidity ^0.8.20;
+
+import {IGovernor, Governor} from "../Governor.sol";
+import {IAuthority} from "../../access/manager/IAuthority.sol";
+import {AuthorityUtils} from "../../access/manager/AuthorityUtils.sol";
+import {IAccessManager} from "../../access/manager/IAccessManager.sol";
+import {IAccessManaged} from "../../access/manager/IAccessManaged.sol";
+import {Address} from "../../utils/Address.sol";
+import {Math} from "../../utils/math/Math.sol";
+import {SafeCast} from "../../utils/math/SafeCast.sol";
+import {Time} from "../../utils/types/Time.sol";
+
+/**
+ * @dev TODO
+ */
+abstract contract GovernorTimelockAccess is Governor {
+    struct ExecutionPlan {
+        uint32 delay;
+        IAccessManager[] managers;
+    }
+
+    mapping(uint256 => ExecutionPlan) private _executionPlan;
+    mapping(address target => address) private _authorityOverride;
+
+    uint32 _baseDelay;
+
+    error GovernorUnmetDelay(uint256 proposalId, uint256 neededTimestamp);
+
+    event BaseDelaySet(uint32 oldBaseDelaySeconds, uint32 newBaseDelaySeconds);
+
+    constructor(uint32 initialBaseDelay) {
+        _baseDelay = initialBaseDelay;
+    }
+
+    /**
+     * @dev Base delay that will be applied to all function calls. Some may be further delayed by their associated
+     * `AccessManager` authority; in this case the final delay will be the maximum of the base delay and the one
+     * demanded by the authority.
+     *
+     * NOTE: Execution delays are processed by the `AccessManager` contracts, and according to that contract are
+     * expressed in seconds. Therefore, the base delay is also in seconds, regardless of the governor's clock mode.
+     */
+    function baseDelaySeconds() public view virtual returns (uint32) {
+        return _baseDelay;
+    }
+
+    /**
+     * @dev Change the value of {baseDelaySeconds}. This operation can only be invoked through a governance proposal.
+     */
+    function setBaseDelaySeconds(uint32 newBaseDelay) public virtual onlyGovernance {
+        _setBaseDelaySeconds(newBaseDelay);
+    }
+
+    /**
+     * @dev Change the value of {baseDelaySeconds}. Internal function without access control.
+     */
+    function _setBaseDelaySeconds(uint32 newBaseDelay) internal virtual {
+        emit BaseDelaySet(_baseDelay, newBaseDelay);
+        _baseDelay = newBaseDelay;
+    }
+
+    /**
+     * @dev Public accessor to check the execution details.
+     */
+    function proposalExecutionPlan(uint256 proposalId) public view returns (ExecutionPlan memory) {
+        return _executionPlan[proposalId];
+    }
+
+    /**
+     * @dev See {IGovernor-propose}
+     */
+    function propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description
+    ) public virtual override returns (uint256) {
+        uint256 proposalId = super.propose(targets, values, calldatas, description);
+
+        uint32 maxDelay = baseDelaySeconds();
```

Comment:
If I'm interpreting correctly this should be the min delay

```suggestion
        uint32 minDelay = baseDelaySeconds();
```

Reply:
But we are using `Math.max` below. What makes you think it should be min?

Reply:
Oh I see, your reasoning is the same as mine, but you read the name as "the minimum delay we need to wait", whereas I've been reading it as "the maximum delay required for any operation".

I think I prefer `maxDelay` because it is consistent with `Math.max`. What do you think?

Reply:
Actually, my rule of thumb is to name the fixed variable as the opposite of the operation.
Let's imagine X is constant and Y is variable

- `Math.max(X, Y)` goes for the max value, which can't be *less* than X
- `Math.min(X, Y)` goes for the min value, which can't be *more* than X

>  I've been reading it as "the maximum delay required for any operation"

Can you elaborate on this? I can't follow why is the maximum delay required, it'd mean that less delay is possible, is that the case?

Reply:
Less delay may be possible for a subset of the proposal operations. But the whole package, atomically, can only be executed after all operations' delays are satisfied. That means we have to wait the max. But I think we agree on this because you haven't questioned the use of `Math.max`.

Reply:
I don't think it's a good idea to rename the variable to `minDelay` because `minDelay = Math.max(minDelay, otherDelay)` seems wrong to me

I am open to renaming to `neededDelay` though.